---
title: "VCF_stats"
output: html_document
author: "Anik Dutta"
date: "23 March 2022"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


## All the stats were produced from a file containing 35 million SNPs from 99 *S. chilense* plants using VCFtools

```{r}
library(ggplot2)
library(tidyverse)
```

# Variant Quality
```{r cars}
var_qual <- read_delim("site_qual.lqual", delim = "\t",
                       col_names = c("chr", "pos", "qual"), skip = 1)
a <- ggplot(var_qual, aes(qual)) + geom_density(fill = "lightsalmon1", colour = "black", alpha = 0.3)
a + theme_light()
```


```{r}
var_qual <- read_delim("site_qual.lqual", delim = "\t",
                       col_names = c("chr", "pos", "qual"), skip = 1)
a <- ggplot(var_qual, aes(qual)) + geom_density(fill = "lightsalmon1", colour = "black", alpha = 0.3)
a + theme_light()+xlim(0, 10000)
```

#### The quality scores of SNPs are actually very high for our sites. A quality score of 30 represents a 1 in 1000 chance that our SNP call is erroneous. Clearly most sites exceed this as I have used this filter in GATK. This suggests that we have a lot of high confidence calls. This is most probably because we have sufficient read depth (as you will see in the next section). However since most sites have a high quality we can see that filtering on this is not really going to be that useful.

# SNP depth
```{r}
var_depth <- read_delim("mean_depth_site.ldepth.mean", delim = "\t",
                        col_names = c("chr", "pos", "mean_depth", "var_depth"), skip = 1)
a <- ggplot(var_depth, aes(mean_depth)) + geom_density(fill = "dodgerblue1", colour = "black", alpha = 0.3)
a + theme_light()

a <- ggplot(var_depth, aes(mean_depth)) + geom_density(fill = "dodgerblue1", colour = "black", alpha = 0.3)
a + theme_light() + xlim(0, 100)

summary(var_depth$mean_depth)
```

#### There are few SNPs with extremely high coverage. As the outliers show, some regions clearly have extremely high coverage and this likely reflects mapping/assembly errors and also paralogous or repetitive regions. I will exclude these as they will bias downstream analyses.

# SNP missingness
```{r}
var_miss <- read_delim("miss_site.lmiss", delim = "\t",
                       col_names = c("chr", "pos", "nchr", "nfiltered", "nmiss", "fmiss"), skip = 1)

a <- ggplot(var_miss, aes(fmiss)) + geom_density(fill = "darkmagenta", colour = "black", alpha = 0.3)
a + theme_light()+xlab("SNP missingness")

summary(var_miss$fmiss)
```

#### The missingness analysis is very good. Most SNPs have no missing data with some exceptions in the Max section of the summary. I initially used a missingness threshold of 50% meaning that it removed all sites where over 50% of individuals are missing a genotype.

# Mean depth per individual
```{r}
ind_depth <- read_delim("depth_ind.idepth", delim = "\t",
                        col_names = c("ind", "nsites", "depth"), skip = 1)  
a <- ggplot(ind_depth, aes(depth)) + geom_histogram(fill = "orange", colour = "black", alpha = 0.3)
a + theme_light()+xlab("Depth per individual")
```

#### While some individuals were sequenced to a lower depth than others, there are no extreme outliers. Therefore, it might indicate that there is no issue with individual sequencing depth.

# Individual Heterozygosity
```{r}
ind_het <- read_delim("het.het", delim = "\t",
                      col_names = c("ind","ho", "he", "nsites", "f"), skip = 1)
a <- ggplot(ind_het, aes(f)) + geom_histogram(fill = "tomato3", colour = "black", alpha = 0.3)
a + theme_light()+xlab("Heterozygosity")
```

#### This is where I don't know what to say. I guess it is expect to observe high heterozygosity in some individuals. Remco, would you weigh in?

# Based on these statistics, I will apply some more filters and see how it affects the analyses.
